### PR TITLE
VP-5640: Forward the scheme for Linux and non-IIS reverse proxies

### PR DESCRIPTION
**Pull request by trueboroda**
According to doc 
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-3.1#forward-the-scheme-for-linux-and-non-iis-reverse-proxies
It allow us generate the right links inside the storefront at proxy server usage.

### CODE REVIEW

File: VirtoCommerce.Storefront/Startup.cs [hunk 1]
```diff
@@ -89,6 +91,22 @@ public void ConfigureServices(IServiceCollection services)
 
             services.Configure<StorefrontOptions>(Configuration.GetSection("VirtoCommerce"));
 
+            if (string.Equals(
```
Comment (asvishnyakov) [inline VirtoCommerce.Storefront/Startup.cs:94]: But we already has `ConfigureWebHostDefaults` in Program.cs. I think it should work when `ASPNETCIRE_FORWARDEDHEADERS_ENABLED` specified without this code, out of the box:
https://stackoverflow.com/a/58908072
https://github.com/dotnet/aspnetcore/blob/1480b998660d2f77d0605376eefab6a83474ce07/src/DefaultBuilder/src/WebHost.cs#L240

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: VirtoCommerce.Storefront/Startup.cs [hunk 2]
```diff
@@ -89,6 +91,22 @@ public void ConfigureServices(IServiceCollection services)
...
+                services.Configure<ForwardedHeadersOptions>(options =>
+                {
+                    options.ForwardedHeaders =
+                        ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+                    // Only loopback proxies are allowed by default.
+                    // Clear that restriction because forwarders are enabled by explicit 
+                    // configuration.
+                    options.KnownNetworks.Clear();
+                    options.KnownProxies.Clear();
+                });
```
Comment (asvishnyakov) [inline VirtoCommerce.Storefront/Startup.cs]: But we already has `ConfigureWebHostDefaults` in Program.cs. I think it should work when `ASPNETCIRE_FORWARDEDHEADERS_ENABLED` specified without this code, out of the box:
https://stackoverflow.com/a/58908072
https://github.com/dotnet/aspnetcore/blob/1480b998660d2f77d0605376eefab6a83474ce07/src/DefaultBuilder/src/WebHost.cs#L240

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: VirtoCommerce.Storefront/Startup.cs (hunk 2 above)
Comment (trueboroda) [reply VirtoCommerce.Storefront/Startup.cs]: Will be done.

Comment (trueboroda) [reply VirtoCommerce.Storefront/Startup.cs]: https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/

**Review by pushnitsa — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**